docs/src/main/sphinx/optimizer/cost-based-optimizations.rst

```diff
@@ -73,14 +76,50 @@ The valid values are:
 Capping replicated table size
 -----------------------------
 
-Join distribution type will be chosen automatically when join reordering strategy
-is set to ``AUTOMATIC`` or when join distribution type is set to ``AUTOMATIC``.
-In such case it is possible to cap the maximum size of replicated table via
-``join-max-broadcast-table-size`` config property (e.g. ``join-max-broadcast-table-size=100MB``)
-or via ``join_max_broadcast_table_size`` session property (e.g. ``set session join_max_broadcast_table_size='100MB';``)
-This allows to improve cluster concurrency and to prevent bad plans when CBO misestimates size of joined tables.
+Join distribution type is automatically chosen when join reordering strategy is
+set to ``AUTOMATIC`` or when join distribution type is set to ``AUTOMATIC``. In
+such cases, it is possible to cap the maximum size of the replicated table with
+the ``join-max-broadcast-table-size`` configuration property or with the
+``join_max_broadcast_table_size`` session property. This allows you to improve
+cluster concurrency and prevent bad plans when the cost-based optimizer
+misestimates size of joined tables.
+
+By default, the replicated table size is capped to 100MB.
+
+Syntactic join order
+--------------------
+
+If not using cost-based optimization, Trino defaults to syntactic join ordering.
+While there is no formal way to optimize queries for this case, it is possible
+to take advantage of how Trino implements joins to make them more performant.
+
+Trino uses in-memory hash joins. When processing a join statement, Trino loads
+the right-most table of the join into memory as the build side, then streams the
+next right-most table as the probe side to execute the join. If a query has
+multiple joins, the result of this first join stays in memory as the build side,
+and the third right-most table is then used as the probe side, and so on for
+additional joins.
+
+Because of this behavior, it is optimal to syntactically order joins from the
+largest tables to the smallest, as this minimizes memory usage.
+
+As an example, if you have a small, medium, and large table and are using left
+joins:
+
+.. code-block:: sql
+
+    SELECT
+      *
+    FROM
+      large_table l
+      LEFT JOIN medium_table m ON l.id = m.id
```

Comment:
using `id` in all three tables is not logically .. maybe  l.l_id = m.l_id  or so ..